src/middlewared/middlewared/plugins/disk_/configure_swap.py

```diff
@@ -0,0 +1,223 @@
+                try:
+                    await self.middleware.call(
+                        'disk.create_mirror', name, {
+                            'paths': [part_a_path, part_b_path],
+                            'extra': {'level': 1} if IS_LINUX else {},
+                        }
+                    )
+                except CallError:
+                    self.logger.warning('Failed to create swap mirror %s: %s', name)
+                    continue
+
+                swap_device = os.path.realpath(os.path.join('/dev/md', name)) if IS_LINUX else f'mirror/{name}'
+                create_swap_devices[swap_device] = {
+                    'path': swap_device,
+                    'encrypted_provider': None,
+                }
+
+            # Add remaining partitions to unused list
+            unused_partitions += partitions
+
+        if unused_partitions and not create_swap_devices and all(
+            not existing_swap_devices[k] for k in existing_swap_devices
+        ):
+            if not IS_LINUX and not dumpdev:
+                await self.middleware.call('disk.dumpdev_configure', unused_partitions[0])
+            swap_device = all_partitions[unused_partitions[0]]['path'] if IS_LINUX else unused_partitions[0]
+            create_swap_devices[swap_device] = {
+                'path': swap_device,
+                'encrypted_provider': all_partitions[unused_partitions[0]]['encrypted_provider'],
+            }
+
+        created_swap_devices = []
+        for name, data in create_swap_devices.items():
+            if IS_LINUX:
+                if not data['encrypted_provider']:
+                    cp = await Popen(
+                        f'echo "y" | cryptsetup -d /dev/urandom open --type plain {name} {name.split("/")[-1]}',
+                        shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                    )
+                    stdout, stderr = await cp.communicate()
+                    if cp.returncode:
+                        self.logger.warning('Failed to encrypt %s device: %s', name, stderr)
+                        continue
+                    name = os.path.join('/dev/mapper', name.split('/')[-1])
+                else:
+                    name = data['encrypted_provider']
+                try:
+                    await run('mkswap', name)
+                except subprocess.CalledProcessError as e:
+                    self.logger.warning(f'Failed to make swap for %s: %s', name, e.stderr.decode())
+                    continue
+            elif not IS_LINUX and not data['encrypted_provider']:
+                try:
+                    await run('geli', 'onetime', name)
+                except subprocess.CalledProcessError as e:
+                    self.logger.warning('Failed to encrypt swap partition %s: %s', name, e.stderr.decode())
+                    continue
+
+            name = name if IS_LINUX else f'{name}.eli'
```

Comment:
if not IS_LINUX:
   name = f'{name}.eli'

reads better, than assigning name to itself (name = name)